packages/frontend/@n8n/design-system/src/v2/components/Tooltip/Tooltip.vue

```diff
@@ -0,0 +1,162 @@
+<script setup lang="ts">
+import {
+	TooltipProvider,
+	TooltipRoot,
+	TooltipTrigger,
+	TooltipContent,
+	TooltipPortal,
+	TooltipArrow,
+} from 'reka-ui';
+import { computed, ref, watch } from 'vue';
+
+import type { N8nTooltipProps, N8nTooltipEmits } from './Tooltip.types';
+import N8nButton from '../../../components/N8nButton';
+import { useInjectTooltipAppendTo } from '../../../composables/useTooltipAppendTo';
+
+defineOptions({
+	inheritAttrs: false,
+});
+
+const props = withDefaults(defineProps<N8nTooltipProps>(), {
+	placement: 'top',
+	showAfter: 0,
+	enterable: true,
+	teleported: true,
+	showArrow: true,
+	justifyButtons: 'flex-end',
+	buttons: () => [],
+	offset: 6,
+});
+
+const emit = defineEmits<N8nTooltipEmits>();
```

Comment:
Our existing `Tooltip` component doesn't emit anything, so it seems this is a new feature we'd be adding on the reka implementation.
What's the use case for emitting an event when the tooltip becomes visible?